clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -78,50 +77,69 @@ abstract class OpenstackClientProvider {
   }
 
   /**
-   * Create or update a security group, applying a list of rules. If the securityGroupId is provided, updates an existing
-   * security group, else creates a new security group.
-   *
-   * Note: 2 default egress rules are created when creating a new security group
-   * automatically with remote IP prefixes 0.0.0.0/0 and ::/0.
-   *
-   * @param securityGroupId id of an existing security group to update
-   * @param securityGroupName name security group
-   * @param description description of the security group
-   * @param rules list of rules for the security group
+   * Creates a security group rule.
+   * @param securityGroupId id of the security group which this rule belongs to
+   * @param protocol the protocol of the rule
+   * @param cidr the cidr for the rule
+   * @param fromPort the fromPort for the rule
+   * @param toPort the toPort for the rule
+   * @return the created rule
    */
-  void upsertSecurityGroup(String securityGroupId, String securityGroupName, String description, List<OpenstackSecurityGroupDescription.Rule> rules) {
-
+  SecGroupExtension.Rule createSecurityGroupRule(String securityGroupId, IPProtocol protocol, String cidr, int fromPort, int toPort) {
     handleRequest(AtomicOperations.UPSERT_SECURITY_GROUP) {
```

Comment:
Are you planning on updating with new handleRequest method without operation name?